### ISSUE TITLE
Initialise default attributes

### ISSUE THREAD

**Issue by simonaberry** [ParsePlatform/Parse-SDK-JS#322]
What is the correct way to set defaults on attributes inside the initialize function 

This issue was [originally raised ](https://github.com/ParsePlatform/Parse-SDK-JS/issues/73#issuecomment-153525815)in  #73 (also #216), and the suggested solution was to use `set()`.

However if i use `this.set('fieldname', value);` it is not working.... do I have the incorrect format ?

Also - assuming this community has access to the official documentation, it would help to update it (happy to help)

**Comment by andrewimm** [comment]
Using the initialize() function is highly discouraged. It's supported for legacy reasons, but you should consider it deprecated.

Here's how I would recommend implementing default attributes, with ES6 classes:

```js
class MyObject extends Parse.Object {
  constructor(arguments) {
    super('MyClassName');

    this.set({
      // default values
    });
    if (arguments) {
      this.set(arguments);
    }
  }

  // ... rest of the class
}
```

(the arguments bit is so that you can still specify initial values to override the defaults)

**Comment by simonaberry** [comment]
Thanks for the clarification... do you have a non-ES6 solution ?

**Comment by andrewimm** [comment]
Sorry, I encourage developers to adopt ES6 classes in their workflow. Can you show me how you attempted to implement `initialize()`? I just threw together a simple example which seems to work fine.

**Comment by simonaberry** [comment]
I guess its the fear of the unknown... id did a quick look at ES6 compatibility in browsers and was worried that your ES6 solution may not work in older clients.... but maybe a shim is in order ?

there is the full (angularjs) code for the factory in question:

```
.factory('FPProject', ['$q', function ($q) {
	var FPProject = Parse.Object.extend('Project', {
	    // Instance methods
	    initialize: function (attrs, options) {
			//set company if passed
			if ((attrs !== undefined) && (attrs.company !== undefined)) {
				this.set('company', attrs.company);
			}
            this.set('name' , 'New Project');
            this.set('status' , 'a'); //active
		}
	},
	{
	     // Class methods			
	});

	// getters & setters
	['name', 'status', 'company'].forEach(function (item) {
			Object.defineProperty(FPProject.prototype, item, {
				get: function () {
					return this.get(item);
				},
				set: function (aValue) {
					return this.set(item, aValue);
				}
			});
		});

	return FPProject;
}]) 
```

thanks for the help!

**Comment by andrewimm** [comment]
Okay, I think I figured out what's going on. #81 was intended to fix an issue with `initialize`, but it broke setting attributes at init time for non-Node.js environments only.

I'll see if I can fix this and push a new version to npm

**Comment by simonaberry** [comment]
fixed in 03d14a0
